lms/djangoapps/courseware/user_state_client.py

```diff
@@ -193,6 +193,11 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
         else:
             user = User.objects.get(username=username)
 
+        if user.is_anonymous():
```

Comment:
I think `user.is_anonymous()` is the right thing here, especially given @cpennington's previous comment about the purpose of `user.known`.  This is specifically to work around the problem of `AnonymousUser`s not having a database id.